### PR TITLE
JSONP response formatter

### PR DESCRIPTION
**Pull request by mdmunir**
| Q | A |
| --- | --- |
| Is bugfix? | yes |
| New feature? | yes/no |
| Breaks BC? | yes |
| Tests pass? | yes |
| Fixed issues | #10943 |

### CODE REVIEW

**Comment by samdark** [comment]
Looks OK. Need upgrade note since it breaks compatibility.

**Comment by SilverFire** [comment]
:+1:

File: framework/UPGRADE.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ Upgrade to Yii 2.1.0
   `yii\db\QueryBuilderbuild::buildGroupBy($columns)` -> `buildGroupBy($columns, &$params)`
   `yii\db\QueryBuilderbuild::buildOrderByAndLimit($sql, $orderBy, $limit, $offset)` -> `buildOrderByAndLimit($sql, $orderBy, $limit, $offset, &$params)`
 
+* The JSONP response formatter have changed. Jsonp callback should be provided via query param or setting up at `\yii\web\Response::formats['jsonp']`.
+  Then the data provided to `\yii\web\Response` has format like json response as.
```
Comment (samdark) [inline framework/UPGRADE.md]: json → JSON

Comment (samdark) [inline framework/UPGRADE.md]: has format like json response as → is formatted as JSON.

File: framework/UPGRADE.md [hunk 2]
```diff
@@ -33,6 +33,8 @@ Upgrade to Yii 2.1.0
   `yii\db\QueryBuilderbuild::buildGroupBy($columns)` -> `buildGroupBy($columns, &$params)`
   `yii\db\QueryBuilderbuild::buildOrderByAndLimit($sql, $orderBy, $limit, $offset)` -> `buildOrderByAndLimit($sql, $orderBy, $limit, $offset, &$params)`
 
+* The JSONP response formatter have changed. Jsonp callback should be provided via query param or setting up at `\yii\web\Response::formats['jsonp']`.
```
Comment (samdark) [inline framework/UPGRADE.md]: Jsonp → JSONP

Comment (samdark) [inline framework/UPGRADE.md]: or setting up at → or configured via

File: framework/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ Yii Framework 2 Change Log
 - Removed methods marked as deprected in 2.0.x (samdark)
 - Chg #11397: Minimum required version of PHP is 5.5.0 now (samdark)
 - Chg: Removed `yii\base\Object::className()` in favor of native PHP syntax `::class`, which does not trigger autoloading (cebe)
+- Chg #10943: Refactoring JSONP response formatter. Only need `data` value. Callback provided via query param (mdmunir)
```
Comment (samdark) [inline framework/CHANGELOG.md]: Refactoring → refactored

**Review by dmirogin — APPROVED** [review]

**Comment by samdark** [comment]
Won't make it into 2.0. Issue was transferred to yiisoft/yii-web.

**Pull request closed without merging**